src/v/model/tests/record_batch_test.cc

```diff
@@ -78,3 +78,15 @@ SEASTAR_THREAD_TEST_CASE(set_max_timestamp) {
     BOOST_TEST(crc == batch.header().crc);
     BOOST_TEST(hdr_crc == batch.header().header_crc);
 }
+
+SEASTAR_THREAD_TEST_CASE(iterator) {
```

Comment:
nit: might add a test for the exceptional case

Reply:
Done.